Twister failing due to missing nrf-regtool

<!--
**Notes**
Github Discussions (https://github.com/zephyrproject-rtos/zephyr/discussions)
are available to first verify that the issue is a genuine Zephyr bug and not a
consequence of Zephyr services misuse.

This issue list is only for bugs in the main Zephyr code base
(https://github.com/zephyrproject-rtos/zephyr/). If the bug is for a project
fork (such as NCS) specific feature, please open an issue in the fork project
instead.
-->

**Describe the bug**
<!--
A clear and concise description of what the bug is.

Please also mention any information which could help others to understand
the problem you're facing:
 - What target platform are you using?
 - What have you tried to diagnose or workaround this issue?
 - Is this a regression? If yes, have you been able to "git bisect" it to a
   specific commit?
 - ...
-->

**To Reproduce**
<!--
Steps to reproduce the behavior:
1. mkdir build; cd build
2. cmake -DBOARD=board\_xyz
3. make
4. See error
-->

Steps to reproduce the behavior:
1. `twister --all -s sample.drivers.crypto.mbedtls`
2. Encounter various build errors for `nrf54h20dk` and `nrf9280pdk` boards:
  ```
  INFO    - The following issues were found (showing the top 10 items):
INFO    - 1) samples/drivers/crypto/sample.drivers.crypto.mbedtls on nrf9280pdk/nrf9280/cpuapp error (Cmake build failure)
INFO    - 2) samples/drivers/crypto/sample.drivers.crypto.mbedtls on nrf9280pdk/nrf9280/cpuppr/xip error (Cmake build failure)
INFO    - 3) samples/drivers/crypto/sample.drivers.crypto.mbedtls on nrf9280pdk/nrf9280/cpurad error (Cmake build failure)
INFO    - 4) samples/drivers/crypto/sample.drivers.crypto.mbedtls on nrf54h20dk/nrf54h20/cpuapp error (Cmake build failure)
INFO    - 5) samples/drivers/crypto/sample.drivers.crypto.mbedtls on nrf54h20dk/nrf54h20/cpuppr error (Cmake build failure)
INFO    - 6) samples/drivers/crypto/sample.drivers.crypto.mbedtls on nrf54h20dk/nrf54h20/cpuppr/xip error (Cmake build failure)
INFO    - 7) samples/drivers/crypto/sample.drivers.crypto.mbedtls on nrf54h20dk/nrf54h20/cpurad error (Cmake build failure)
INFO    - 8) samples/drivers/crypto/sample.drivers.crypto.mbedtls on nrf54h20dk/nrf54h20/cpuflpr error (Cmake build failure)
INFO    - 9) samples/drivers/crypto/sample.drivers.crypto.mbedtls on nrf54h20dk/nrf54h20/cpuflpr/xip error (Cmake build failure)
  ```

**Expected behavior**
<!--
A clear and concise description of what you expected to happen.
-->

Running twister successfully on all supported architectures without needing to install extra/undocumented dependencies.

**Impact**
<!--
What impact does this issue have on your progress (e.g., annoyance, showstopper)
-->

Can not run Twister for all architecture without encountering errors.

**Logs and console output**
<!--
If applicable, add console logs or other types of debug information
e.g Wireshark capture or Logic analyzer capture (upload in zip archive).
copy-and-paste text and put a code fence (\`\`\`) before and after, to help
explain the issue. (if unable to obtain text log, add a screenshot)
-->

The errors in build.log files look like this:

```
CMake Error at <snip>/zephyr/modules/hal_nordic/CMakeLists.txt:15 (find_package):
  Could not find a configuration file for package "nrf-regtool" that is
  compatible with requested version "5.6.0".

  The following configuration files were considered but not accepted:

    <snip>/modules/hal_nordic/nrf-regtool/nrf-regtoolConfig.cmake, version: unknown
```

**Environment (please complete the following information):**

 - OS: Linux
 - Toolchain: Zephyr SDK 0.16.8
 - Commit: 44d101acabb5f3c0916ba9c193361c32e16c015c


**Workaround**

* `pip install nrf-regtool`

--- a/scripts/west_commands/runners/nrf_common.py
+++ b/scripts/west_commands/runners/nrf_common.py
@@ -35,9 +35,17 @@ UICR_RANGES = {
         'NRFDL_DEVICE_CORE_APPLICATION': (0x00FF8000, 0x00FF8800),
         'NRFDL_DEVICE_CORE_NETWORK': (0x01FF8000, 0x01FF8800),
     },
+    'NRF54H_FAMILY': {
+        'NRFDL_DEVICE_CORE_APPLICATION': (0x0FFF8000, 0x0FFF8800),
+        'NRFDL_DEVICE_CORE_NETWORK': (0x0FFFA000, 0x0FFFA800),
+    },
     'NRF91_FAMILY': {
         'NRFDL_DEVICE_CORE_APPLICATION': (0x00FF8000, 0x00FF8800),
-    }
+    },
+    'NRF92_FAMILY': {
+        'NRFDL_DEVICE_CORE_APPLICATION': (0x0FFF8000, 0x0FFF8800),
+        'NRFDL_DEVICE_CORE_NETWORK': (0x0FFFA000, 0x0FFFA800),
+    },
 }
 
 # Relative to the root of the hal_nordic module
@@ -307,6 +315,19 @@ class NrfBinaryRunner(ZephyrBinaryRunner):
                 self.build_conf.getboolean('CONFIG_SOC_NRF54H20_ENGB_CPURAD') or
                 self.build_conf.getboolean('CONFIG_SOC_NRF9280_CPURAD')
             )
+            generated_uicr = self.build_conf.getboolean('CONFIG_NRF_REGTOOL_GENERATE_UICR')
+
+            if cpuapp:
+                core = 'NRFDL_DEVICE_CORE_APPLICATION'
+            elif cpurad:
+                core = 'NRFDL_DEVICE_CORE_NETWORK'
+
+            if generated_uicr and not self.hex_get_uicrs().get(core):
+                raise RuntimeError(
+                    f"Expected a UICR to be contained in: {self.hex_}\n"
+                    "Please ensure that the correct version of nrf-regtool is "
+                    "installed, then run 'west build --cmake' to try again."
+                )
 
             if self.erase:
                 self.exec_op('erase', core='NRFDL_DEVICE_CORE_APPLICATION')
@@ -335,18 +356,9 @@ class NrfBinaryRunner(ZephyrBinaryRunner):
                         mpi_hex_dir / 'suit_installed_envelopes_application_merged.hex')
                     self.op_program(app_root_envelope_hex_file, 'ERASE_NONE', None, defer=True, core='NRFDL_DEVICE_CORE_APPLICATION')
 
-            if cpuapp:
-                if not self.erase and self.build_conf.getboolean('CONFIG_NRF_REGTOOL_GENERATE_UICR'):
-                    self.exec_op('erase', core='NRFDL_DEVICE_CORE_APPLICATION',
-                                 option={'chip_erase_mode': 'ERASE_UICR',
-                                         'qspi_erase_mode': 'ERASE_NONE'})
-                core = 'NRFDL_DEVICE_CORE_APPLICATION'
-            elif cpurad:
-                if not self.erase and self.build_conf.getboolean('CONFIG_NRF_REGTOOL_GENERATE_UICR'):
-                    self.exec_op('erase', core='NRFDL_DEVICE_CORE_NETWORK',
-                                 option={'chip_erase_mode': 'ERASE_UICR',
-                                         'qspi_erase_mode': 'ERASE_NONE'})
-                core = 'NRFDL_DEVICE_CORE_NETWORK'
+            if not self.erase and generated_uicr:
+                self.exec_op('erase', core=core, option={'chip_erase_mode': 'ERASE_UICR',
+                                                         'qspi_erase_mode': 'ERASE_NONE'})
         else:
             if self.erase:
                 erase_arg = 'ERASE_ALL'

